spec.bs

```diff
@@ -59,16 +60,250 @@ A <dfn>import map</dfn> is a [=struct=] with two [=struct/items=]:
+   asynchronously complete the algorithm with a failure.
+1. <a spec="html">Spin the event loop</a> until |settings object|'s
+   [=environment settings object/pending import maps count=] is 0.
+1. Set |settings object|'s
+   [=environment settings object/acquiring import maps flag=] to false.
+1. Asynchronously complete the algorithm with a success.
+
+</div>
+
+Insert a call to [=wait for import maps=] at the beginning of the following HTML spec concepts,
+and complete the algorithms with a failure when [=wait for import maps=] completes with a failure.
+
+- <a spec="html">fetch a module script graph</a>
+- fetch a dynamic module script graph
+- fetch an inline module script graph
+- <a spec="html">fetch a module worker script graph</a> using
+  <var ignore>module map settings object</var>
+
+Insert a call to [=wait for import maps=] before Step 11
+(a call to <a spec="html">fetch a single module script</a>) of
+<a spec="html">fetch and process the linked resource</a> algorithm for modulepreload links, and
+skip Step 11 and set <var ignore>result</var> to null
+when [=wait for import maps=] completes with a failure.
+
+<p class="advisement">
+In the current HTML-based spec,
+the settings object used here is |module map settings object|,
+not fetch client settings object, because
+[=resolve a module specifier=] uses
+the import map of |module map settings object|.
+In fetch-spec-based import map, this would be
+<var ignore>fetch client settings object</var> instead.
+This only affects <a spec="html">fetch a module worker script graph</a>
+where these two settings objects are different.
+</p>
+
+<p class="advisement">Depending on the exact location of
+[=wait for import maps=], there would be observable behavior differences
+between HTML- and Fetch-based import maps.
+For example, the current HTML-spec-based draft here calls
+[=wait for import maps=] at very early stages of module loading before
+[=resolve a module specifier=]. Therefore,
+[=environment settings object/acquiring import maps flag=] is always
+cleared, even if [=resolve a module specifier=] fails e.g. in parsing
+http://:invalid.
+When we switch to Fetch-spec-based, import map resolution would be
+done inside Fetch spec after [=resolve a module specifier=] and thus
+we might also want to [=wait for import maps=] after
+[=resolve a module specifier=].
+In that case,
+[=environment settings object/acquiring import maps flag=] wouldn't
+be cleared for http://:invalid, because we early-exit module loading.
+</p>
+
+<h3 id="integration-register-an-import-map">Registering an import map</h3>
 
-<!-- TODO unexport -->
+<div algorithm>
+To <dfn>register an import map</dfn> given a [=string=] or a failure |source text|,
+|settings object|, a [=URL=] |base URL|, and an [=/Element=] |element|,
+run the following steps.
+1. If |source text| is a failure:
+    1. <a spec="html">queue a task</a> to <a spec="html">fire an event</a> named `error` at
+       |element|, and return.
```

Comment:
Nit: this algorithm has a lot of single-step nested lists. Instead of `1. If x, \n  1. y` just say `If x, then y`, on one line.

Reply:
Done.